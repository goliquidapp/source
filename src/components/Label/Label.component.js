import React, {Component} from 'react';
import { View, Text } from 'react-native';
import Colors from '../../resources/Colors.js';

import {Icon} from 'react-native-elements'

class Label extends Component{
	render(){
		const {containerStyle, textStyle, titleStyle, icon, type}=this.props
		return (
			<View style={[styles.container,containerStyle]}>
				<Icon name={icon} type={type} size={18}/>
				<Text style={[styles.title,titleStyle]}>{this.props.title}</Text>
				<Text style={[styles.text,textStyle]}>{this.props.text}</Text>
			</View>
		)
	}
}

const styles={
	container:{
		flex:1,
		flexDirection:'row',
		alignItems:'center',
		justifyContent:'center',
		alignSelf:'center',
		width:'90%',
		marginVertical:5
	},
	text:{
		color:Colors['Black Pearl'],
		fontWeight:'bold',
		fontSize:16,
		width:'60%'
	},
	title:{
		color:Colors['Black Pearl'],
		fontSize:16,
		width:'20%',
		marginLeft:10
	}
}

export default Label;