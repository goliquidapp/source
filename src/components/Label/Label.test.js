import React from 'react';
import { Text, View} from 'react-native';
import renderer from 'react-test-renderer';

import Label from './Label.component';

describe("Testing Label component",()=>{
	it('renders correctly', () => {
		const wrapper = renderer.create(
    		<Label value={""}/>
    	).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
})