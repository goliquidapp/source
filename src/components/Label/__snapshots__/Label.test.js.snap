// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Label component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 5,
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "justifyContent": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
      underlayColor="white"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 18,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#041322",
          "fontSize": 16,
          "marginLeft": 10,
          "width": "20%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#041322",
          "fontSize": 16,
          "fontWeight": "bold",
          "width": "60%",
        },
        undefined,
      ]
    }
  />
</View>
`;
